supervisor/oauth_verifiers.go

```diff
@@ -0,0 +1,62 @@
+package supervisor
+
+import (
+	"github.com/google/go-github/github"
+	"github.com/markbates/goth"
+	"github.com/starkandwayne/goutils/log"
+	"net/http"
+)
+
+var OAuthVerifier MembershipChecker
+
+type MembershipChecker interface {
+	Verify(goth.User, *http.Client) bool
+}
+
+// implement github verifier that takes users, groups, teams, retrieves membership from github library, checks to see if they're configured
+
+type GithubVerifier struct {
+	Orgs []string
+}
+
+func (gv *GithubVerifier) Verify(user goth.User, c *http.Client) bool {
+	// If none specified, just let any authenticated person in
+	if len(gv.Orgs) == 0 {
+		log.Debugf("No orgs required for authorization, granting access to '%s'.", user.NickName)
```

Comment:
Can we make this deny by default, in case someone forgets to configure their organizations in the deployment?